src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -0,0 +1,1885 @@
+			}
+		}
+
+	}
+
+	/**
+	 * This method is used to control exposure
+	 */
+	private void exposureControl(String value, Command exposureCommand) {
+		switch (exposureCommand) {
+			case BACKLIGHT: {
+				if (value.equals(SWITCH_STATUS_ON)) {
+					backlightControl(BacklightStatus.ON);
+				} else if (value.equals(SWITCH_STATUS_OFF)) {
+					backlightControl(BacklightStatus.OFF);
+				}
+				break;
+			}
+			case AE_MODE: {
+				aeModeControl(AEMode.getByName(value));
+				break;
+			}
+			case EXP_COMP_DIRECT:
+			case GAIN_LIMIT_DIRECT:
+			case GAIN_DIRECT:
+			case IRIS_DIRECT:
+			case SHUTTER_DIRECT: {
+				float directValue = Float.parseFloat(value);
+				directControl(exposureCommand, (int) directValue);
+				break;
+			}
+			case AUTO_SLOW_SHUTTER: {
+				if (value.equals(SWITCH_STATUS_ON)) {
+					slowShutterControl(SlowShutterStatus.ON);
+				} else if (value.equals(SWITCH_STATUS_OFF)) {
+					slowShutterControl(SlowShutterStatus.OFF);
+				}
+				break;
+			}
+			default: {
+				throw new IllegalStateException("Unexpected value: " + exposureCommand);
+			}
+		}
+	}
+
+	/**
+	 * This method is used to control power
+	 */
+	public void powerControl(PowerStatus powerStatus) {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(buildSendPacket(cameraID, currentSeqNum, PayloadType.COMMAND.getCode(), CommandType.COMMAND.getCode(), PayloadCategory.CAMERA.getCode(),
+					Command.POWER.getCode(), powerStatus.getCode()));
+
+			digestResponse(response, currentSeqNum, CommandType.COMMAND, null);
+		} catch (Exception e) {
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("error during power " + powerStatus.getName() + " send", e);
+			}
+		}
+	}
+
+	/**
+	 * This method is used to control zoom
+	 */
+	public void zoomControl(ZoomControl zoomControl) {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(buildSendPacket(cameraID, currentSeqNum, PayloadType.COMMAND.getCode(), CommandType.COMMAND.getCode(), PayloadCategory.CAMERA.getCode(),
+					Command.ZOOM.getCode(), (byte) (zoomControl.getCode() + zoomSpeed)));
+
+			digestResponse(response, currentSeqNum, CommandType.COMMAND, null);
+		} catch (Exception e) {
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("error during zoom " + zoomControl.getName() + " send", e);
+			}
+		}
+
+		zoomStop();
+	}
+
+	/**
+	 * This method is used to control zoom stop
+	 */
+	public void zoomStop() {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(buildSendPacket(cameraID, currentSeqNum, PayloadType.COMMAND.getCode(), CommandType.COMMAND.getCode(), PayloadCategory.CAMERA.getCode(),
+					Command.ZOOM.getCode(), ZoomControl.STOP.getCode()));
+
+			digestResponse(response, currentSeqNum, CommandType.COMMAND, null);
+		} catch (Exception e) {
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("error during zoom stop send", e);
```

Comment:
exception + debug here also, same as above. Please check other similar places not to swallow any exceptions

Reply:
thank you, I will check again